mobly/controllers/monsoon.py

```diff
@@ -904,26 +816,23 @@ def measure_power(self, hz, duration, tag, offset=30):
         num = duration * hz
         oset = offset * hz
         data = None
-        try:
-            self.usb("auto")
-            time.sleep(1)
-            self.dut.stop_services()
+        self.usb("auto")
+        time.sleep(1)
+        with self.dut.handle_device_disconnect():
             time.sleep(1)
-            data = self.take_samples(hz, num, sample_offset=oset)
-            if not data:
-                raise MonsoonError((
-                    "No data was collected in measurement %s.") % tag)
-            data.tag = tag
-            self.log.info("Measurement summary: %s", repr(data))
-        finally:
-            self.mon.StopDataCollection()
-            self.log.info("Finished taking samples, reconnecting to dut.")
-            self.usb("on")
-            self._wait_for_device(self.dut)
-            # Wait for device to come back online.
-            time.sleep(10)
-            self.dut.start_services()
-            # Release wake lock to put device into sleep.
-            self.dut.sl4a.goToSleepNow()
-            self.log.info("Dut reconnected.")
-            return data
+            try:
+                data = self.take_samples(hz, num, sample_offset=oset)
+                if not data:
+                    raise MonsoonError("No data was collected in measurement %s." %
+                                       tag)
+                data.tag = tag
+                self.log.info("Measurement summary: %s", repr(data))
+                return data
+            finally:
+                self.mon.StopDataCollection()
+                self.log.info("Finished taking samples, reconnecting to dut.")
+                self.usb("on")
+                # Wait for device to come back online.
+                time.sleep(10)
```

Comment:
Why do we need a sleep right before wait_for_device? Is the default timeout of wait_for_device not long enough? (can we increase it?)